src/chain/chain.rs

```diff
@@ -1108,37 +1064,41 @@ impl Chain {
     }
 
     /// Returns whether we should split into two sections.
-    fn should_split(&self, members: &BTreeMap<XorName, P2pNode>) -> Result<bool, RoutingError> {
+    fn should_split(&self) -> Result<bool, RoutingError> {
         if self.state.split_in_progress {
             return Ok(false);
         }
 
-        let new_size = members
-            .values()
-            .filter(|p2p_node| {
-                self.our_id
-                    .name()
-                    .common_prefix(p2p_node.public_id().name())
-                    > self.our_prefix().bit_count()
-            })
-            .count();
-        let min_split_size = self.min_split_size();
+        let our_name = self.our_id.name();
+        let our_prefix_bit_count = self.our_prefix().bit_count();
+        let (our_new_size, sibling_new_size) = self
+            .state
+            .our_joined_members()
+            .map(|(name, _)| our_name.common_prefix(name) > our_prefix_bit_count)
+            .fold((0, 0), |acc, is_our_prefix| {
```

Comment:
wondering why not use the previous calculations? say:
```
let our_new_size = self.state.our_joined_members().filter(|node| node.common_prefix(name) ? our_prefix_bit_count).count();
let sibling_new_size = `self.state.our_joined_members().count() - our_new_size ;
```

Reply:
I guess it's to iterate through `our_joined_members` just once instead of twice. I like it :wink:

Reply:
ok then. it's just the fold format is little bit not that straight forward to read at the first glance.

Reply:
Could you be more specific, is it the use of fold in general, or something here?

Reply:
One way could be to have 

```
|(ours, siblings), is_our_prefix| {
                 if is_our_prefix {
                    (ours + 1, siblings)
                } else {
                    (ours, siblings + 1)
                } 
}

Reply:
the part that it is a tuple to be folded.
something.0 and something.1 is always not reading friendly I think?

changing to `ours, siblings` will be much better.